apps/omg/test/omg/ethereum_event_listener/core_test.exs

```diff
@@ -24,165 +24,142 @@ defmodule OMG.EthereumEventListener.CoreTest do
   @service_name :name
   @request_max_size 5
 
-  test "asks until root chain height provided" do
-    create_state(0, request_max_size: 100)
-    |> Core.get_events_range_for_download(%SyncGuide{sync_height: 1, root_chain_height: 10})
+  test "will respect request_max_size if its allowed!" do
+    create_state(0, request_max_size: 10)
+    |> Core.get_events_range(%SyncGuide{sync_height: 20, root_chain_height: 10})
+    |> assert_range({1, 11})
+
+    create_state(0, request_max_size: 10)
+    |> Core.get_events_range(%SyncGuide{sync_height: 11, root_chain_height: 10})
+    |> assert_range({1, 11})
+
+    create_state(0, request_max_size: 10)
+    |> Core.get_events_range(%SyncGuide{sync_height: 10, root_chain_height: 10})
     |> assert_range({1, 10})
   end
 
-  test "max request size respected" do
+  test "max request size is ignored if above allowed sync_height" do
     create_state(0, request_max_size: 2)
-    |> Core.get_events_range_for_download(%SyncGuide{sync_height: 1, root_chain_height: 10})
-    |> assert_range({1, 2})
+    |> Core.get_events_range(%SyncGuide{sync_height: 1, root_chain_height: 10})
+    |> assert_range({1, 1})
   end
 
-  test "max request size ignored if caller is insiting to get a lot of events" do
-    # this might be counterintuitive, but to we require that the requested sync_height is never left unhandled
+  test "max request size ignored if caller is insisting to get a lot of events" do
     create_state(0, request_max_size: 2)
-    |> Core.get_events_range_for_download(%SyncGuide{sync_height: 4, root_chain_height: 10})
-    |> assert_range({1, 4})
+    |> Core.get_events_range(%SyncGuide{sync_height: 4, root_chain_height: 10})
+    |> assert_range({1, 3})
   end
 
   test "works well close to zero" do
-    create_state(0)
-    |> Core.get_events_range_for_download(%SyncGuide{sync_height: 1, root_chain_height: 10})
-    |> assert_range({1, 5})
-    |> Core.add_new_events([event(1), event(3), event(4), event(5)])
-    |> Core.get_events(0)
-    |> assert_events(events: [], check_in_and_db: 0)
-    |> Core.get_events(1)
-    |> assert_events(events: [event(1)], check_in_and_db: 1)
-    |> Core.get_events(2)
-    |> assert_events(events: [], check_in_and_db: 2)
-    |> Core.get_events(3)
-    |> assert_events(events: [event(3)], check_in_and_db: 3)
-  end
+    0
+    |> create_state()
+    # |> IO.inspect
```

Comment:
will you remove it? I think it was used during debugging. There are a couple of `IO.inspect`'s below

Reply:
oh shit, yes